examples/pubsub.rs

```diff
@@ -57,39 +59,39 @@ impl subscriber::FutureService for Subscriber {
 }
 
 impl Subscriber {
-    fn listen(id: u32) -> SocketAddr {
+    fn listen(id: u32, options: server::Options) -> SocketAddr {
         Subscriber { id: id }
-            .listen("localhost:0".first_socket_addr(),
-                    server::Options::default())
-            .wait()
+            .listen("localhost:0".first_socket_addr(), options)
             .unwrap()
     }
 }
 
 #[derive(Clone, Debug)]
 struct Publisher {
-    clients: Arc<Mutex<HashMap<u32, subscriber::FutureClient>>>,
+    clients: Rc<RefCell<HashMap<u32, subscriber::FutureClient>>>,
```

Comment:
Not to resist adding a comment, but I think it's worth pointing out that in Rust thread safety is guaranteed by the compiler, so using `Rc<RefCell<>>` over `Arc<Mutex<>>` is always the right thing as long as the compiler allows it. In this case, `Rc` is not `Send`, which is why I had to remove the `Send` bound before I could replace `Arc`.